src/SFA.DAS.Commitments.Application/Commands/RejectTransferRequest/RejectTransferRequestCommandHandler.cs

```diff
@@ -0,0 +1,129 @@
+using FluentValidation;
+using MediatR;
+using SFA.DAS.Commitments.Application.Exceptions;
+using SFA.DAS.Commitments.Application.Interfaces.ApprenticeshipEvents;
+using SFA.DAS.Commitments.Application.Rules;
+using SFA.DAS.Commitments.Application.Services;
+using SFA.DAS.Commitments.Domain;
+using SFA.DAS.Commitments.Domain.Data;
+using SFA.DAS.Commitments.Domain.Entities;
+using SFA.DAS.Commitments.Domain.Entities.History;
+using SFA.DAS.Commitments.Domain.Interfaces;
+using SFA.DAS.Commitments.Events;
+using SFA.DAS.Messaging.Interfaces;
+
+namespace SFA.DAS.Commitments.Application.Commands.RejectTransferRequest
+{
+    public class RejectTransferRequestCommandHandler: AsyncRequestHandler<RejectTransferRequestCommand>
+    {
+        private readonly AbstractValidator<RejectTransferRequestCommand> _validator;
+        private readonly ICommitmentRepository _commitmentRepository;
+        private readonly ICurrentDateTime _currentDateTime;
+        private readonly IApprenticeshipEventsList _apprenticeshipEventsList;
+        private readonly IApprenticeshipEventsPublisher _apprenticeshipEventsPublisher;
+        private readonly IMessagePublisher _messagePublisher;
+        private readonly CohortApprovalService _cohortApprovalService;
+        private readonly HistoryService _historyService;
+
+        public RejectTransferRequestCommandHandler(AbstractValidator<RejectTransferRequestCommand> validator,
+            ICommitmentRepository commitmentRepository, IApprenticeshipRepository apprenticeshipRepository,
+            IApprenticeshipOverlapRules overlapRules, ICurrentDateTime currentDateTime,
+            IApprenticeshipEventsList apprenticeshipEventsList,
+            IApprenticeshipEventsPublisher apprenticeshipEventsPublisher, IMediator mediator,
+            IMessagePublisher messagePublisher,
+            IHistoryRepository historyRepository)
+        {
+            _validator = validator;
+            _commitmentRepository = commitmentRepository;
+            _currentDateTime = currentDateTime;
+            _apprenticeshipEventsList = apprenticeshipEventsList;
+            _apprenticeshipEventsPublisher = apprenticeshipEventsPublisher;
+            _messagePublisher = messagePublisher;
+            _historyService = new HistoryService(historyRepository);
+
+            _cohortApprovalService = new CohortApprovalService(apprenticeshipRepository, overlapRules, currentDateTime,
+                commitmentRepository, apprenticeshipEventsList, apprenticeshipEventsPublisher, mediator);
+        }
+
+        protected override async Task HandleCore(RejectTransferRequestCommand command)
+        {
+            _validator.ValidateAndThrow(command);
+
+            var commitment = await _commitmentRepository.GetCommitmentById(command.CommitmentId);
+            if (commitment == null)
+            {
+                throw new ResourceNotFoundException();
+            }
+
+            CheckAuthorization(command, commitment);
```

Comment:
CheckAuthorization and CheckCommitmentStatus are identical in both cases. Is it worth creating a abstract base class which both command handlers can inherit from? No the parameter types will be different, so we can't reuse teh code unless we make the command parameter have a common base class